osu.Game.Rulesets.Osu/Objects/Drawables/DrawableOsuBlinds.cs

```diff
@@ -0,0 +1,147 @@
+using osu.Game.Rulesets.Osu.Objects.Drawables.Pieces;
+
+namespace osu.Game.Rulesets.Osu.Objects.Drawables
+{
+    /// <summary>
+    /// Element for the Blinds mod drawing 2 black boxes covering the whole screen which resize inside a restricted area with some leniency.
+    /// </summary>
+    public class DrawableOsuBlinds : Container
+    {
+        /// <summary>
+        /// Black background boxes behind blind panel textures.
+        /// </summary>
+        private Box box1, box2;
+        private Sprite panelLeft, panelRight;
+        private Sprite bgPanelLeft, bgPanelRight;
+        private float target = 1;
+        private readonly float easing = 1;
+        private readonly Container restrictTo;
+        private ISkinSource skin;
+
+        /// <summary>
+        /// <para>
+        /// Percentage of playfield to extend blinds over. Basically moves the origin points where the blinds start.
+        /// </para>
+        /// <para>
+        /// -1 would mean the blinds always cover the whole screen no matter health.
+        /// 0 would mean the blinds will only ever be on the edge of the playfield on 0% health.
+        /// 1 would mean the blinds are fully outside the playfield on 50% health.
+        /// Infinity would mean the blinds are always outside the playfield except on 100% health.
+        /// </para>
+        /// </summary>
+        private const float leniency = 0.2f;
+
+        public DrawableOsuBlinds(Container restrictTo)
+        {
+            this.restrictTo = restrictTo;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ISkinSource skin, TextureStore textures)
+        {
+            RelativeSizeAxes = Axes.Both;
+            Width = 1;
+            Height = 1;
+
+            Add(box1 = new Box
+            {
+                Anchor = Anchor.TopLeft,
+                Origin = Anchor.TopLeft,
+                Colour = Color4.Black,
+                RelativeSizeAxes = Axes.Y,
+                Width = 0,
+                Height = 1
+            });
+            Add(box2 = new Box
+            {
+                Anchor = Anchor.TopRight,
+                Origin = Anchor.TopRight,
+                Colour = Color4.Black,
+                RelativeSizeAxes = Axes.Y,
+                Width = 0,
+                Height = 1
+            });
+            
+            Add(bgPanelLeft = new ModBlindsPanelSprite {
+                Origin = Anchor.TopRight,
+                Colour = Color4.Gray
+            });
+            Add(bgPanelRight = new ModBlindsPanelSprite {
+                Origin = Anchor.TopLeft,
+                Colour = Color4.Gray
+            });
+
+            Add(panelLeft = new ModBlindsPanelSprite {
+                Origin = Anchor.TopRight
+            });
+            Add(panelRight = new ModBlindsPanelSprite {
+                Origin = Anchor.TopLeft
+            });
+
+            this.skin = skin;
+            skin.SourceChanged += skinChanged;
```

Comment:
I'm not sure if I used the skinning here correctly, if I use `skin.GetTexture` here instead of `textures.Get` I don't get the texture I want, but the SkinnableDrawable implements it like this too and I didn't want to use it because I am using the same texture 4 times and don't want to look it up every time so often.